src/frontend/src/contexts/LanguageContext.tsx

```diff
@@ -1,13 +1,45 @@
 import { i18n } from '@lingui/core';
+import { t } from '@lingui/macro';
 import { I18nProvider } from '@lingui/react';
 import { useEffect } from 'react';
 
 import { api } from '../App';
 import { useLocalState } from '../states/LocalState';
 
 // Definitions
-export type Locales = 'en' | 'de' | 'hu' | 'pseudo-LOCALE';
-export const languages: Locales[] = ['en', 'de', 'hu'];
+export type Locales = keyof typeof languages | 'pseudo-LOCALE';
+
+export const languages: Record<string, string> = {
```

Comment:
You need to remove the typing here, so that autocompletion works for the languages obj.
```suggestion
export const languages = {
```

Reply:
How is autocomplete better then typesafety?

Reply:
The type gets automatically interfered from the object definition, if you remove the typing. Basically like your definition, but without the more generic string typing for the keys but with all available keys.